azure-pipelines.yml

```diff
@@ -0,0 +1,169 @@
+    endpoint: "GitHub (SFA)"
+
+steps:
+- task: UseGitVersion@5
+  inputs:
+    versionSpec: 5.x
+    useConfigFile: true
+    configFilePath: GitVersion.yml
+
+- template: azure-pipelines-templates/dependency-check.yml@das-platform-building-blocks
+  
+- task: NuGetToolInstaller@0
+  displayName: 'Use NuGet 4.3.0'
+  inputs:
+    versionSpec: 4.x
+
+- task: NuGetCommand@2
+  displayName: 'NuGet Restore SFA.DAS.ProviderApprenticeshipsService.sln'
+  inputs:
+    restoreSolution: 'src/SFA.DAS.ProviderApprenticeshipsService.sln'
+
+- task: DotNetCoreCLI@2
+  displayName: 'dotnet restore all'
+  inputs:
+    command: restore
+    projects: "src/**/*.csproj"
+
+- task: VSBuild@1
+  displayName: 'Build SFA.DAS.ProviderApprenticeshipsService.sln'
+  inputs:
+    solution: 'src/SFA.DAS.ProviderApprenticeshipsService.sln'
+    platform: '$(buildPlatform)'
+    configuration: '$(buildConfiguration)'
+    clean: true
+
+- task: VSTest@2
+  displayName: 'Run NUnit Tests'
+  inputs:
+    testAssemblyVer2: |
+      **/bin/Release/*UnitTests.dll
+      **/bin/Release/*/*UnitTests.dll
+      !**/*ClientV2.UnitTests.dll
+    platform: '$(buildPlatform)'
+    configuration: '$(buildConfiguration)'    
+
+- task: NuGetCommand@2
+  displayName: 'NuGet Pack Provider Accounts API Client'
+  inputs:
+    command: 'pack'
+    packagesToPack: 'src/SFA.DAS.PAS.Account.Api.Client/SFA.DAS.PAS.Account.Api.Client.csproj'
+    packDestination: '$(Build.ArtifactStagingDirectory)/nuget-publish'
+    versioningScheme: byBuildNumber
+
+- task: VSBuild@1
+  displayName: 'Publish Provider Accounts API'
+  inputs:
+    solution: 'src/SFA.DAS.PAS.Account.Api/SFA.DAS.PAS.Account.Api.csproj'
+    vsVersion: 15.0
+    msbuildArgs: '/p:DeployOnBuild=true /p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true /p:PackageLocation="$(build.artifactstagingdirectory)/Provider Accounts API"'
+    platform: 'anycpu'
+    configuration: '$(buildConfiguration)'
+    clean: true
+
+- task: VSBuild@1
+  displayName: 'Publish Provider Portal'
+  inputs:
+    solution: 'src/SFA.DAS.ProviderApprenticeshipsService.Web/SFA.DAS.ProviderApprenticeshipsService.Web.csproj'
+    vsVersion: 15.0
+    msbuildArgs: '/p:DeployOnBuild=true /p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true /p:PackageLocation="$(build.artifactstagingdirectory)/Provider Portal"'
+    platform: 'anycpu'
+    configuration: '$(buildConfiguration)'
+    clean: true
+
+- task: VSBuild@1
+  displayName: 'Publish Provider WebJob'
+  inputs:
+    solution: 'src/SFA.DAS.PAS.Host.WebJob/SFA.DAS.PAS.Host.WebJob.csproj'
+    vsVersion: 15.0
+    msbuildArgs: '/p:DeployOnBuild=true /p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true /p:PackageLocation="$(build.artifactstagingdirectory)/Provider WebJob"'
+    platform: 'anycpu'
+    configuration: '$(buildConfiguration)'
+    clean: true        
+
+- task: DotNetCoreCLI@2
```

Comment:
Is there a reason why each test has its own task? Is it not possible to run it all under one task and reference multiple projects?

Reply:
I thought of including them all in the same task, but preferred the ability to name each test run separately as each project was also built separately in the original pipeline. This could be a ways of working up for discussion?

Reply:
Updated to combine dotnet test and build tasks to test and build multiple projects per task